pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -884,6 +884,12 @@ func (b *volumeBinder) checkBoundClaims(logger klog.Logger, claims []*v1.Persist
 			return false, true, err
 		}
 
+		err = volume.CheckCSINodeAffinity(csiNode, pv)
```

Comment:
There is another scenario that may need to be considered: after the node joins the cluster, kubelet immediately creates a CSINode. At this time, the csi-plugin has not been started, and the corresponding Driver information is not registered in the CSINode. If there is a new Pod at this time and the PV needs to be dynamically created, it may be scheduled to such a node by the scheduler. At this time, the PV creation may fail, and the scheduler will have to wait for a very long time(ten minutes by default) to complete the Bind operation.

Reply:
yeah, you're right. We should take this scenario into consideration as well~

Reply:
@eahydra I found out during the actual implementation that there is no way to support csinode in dynamic provision scenario. Because the current Kubernetes storage driver is not only CSI, there are other storage drivers that do not have csinode resources.  We can't tell from the storageclass provisioner field whether it's a csi or some other storage driver that's currently providing the service. 

In static binding or pv already exists scenario, because of the presence of the pv, I can tell by ```pv.Spec.CSI``` that the current pv is using the csi driver for declaration cycle management. So it's fine.

Reply:
In the scenario you've described above, I understand that there's nothing better to do than wait.

Reply:
> @eahydra I found out during the actual implementation that there is no way to support csinode in dynamic provision scenario. Because the current Kubernetes storage driver is not only CSI, there are other storage drivers that do not have csinode resources. We can't tell from the storageclass provisioner field whether it's a csi or some other storage driver that's currently providing the service.
> 
> In static binding or pv already exists scenario, because of the presence of the pv, I can tell by `pv.Spec.CSI` that the current pv is using the csi driver for declaration cycle management. So it's fine.

For clarification, consider an instance where a user requests a StorageClass with the provisioner "test.csi.storage.io". However, this provisioner functions as a Flexvolume driver, which means that corresponding CSINode resources will not be present in the cluster.